src/main/ui/app/component/SearchBox.jsx

```diff
@@ -21,43 +22,57 @@ class SearchBox extends Component {
 
     setQuery(query) {
         this.setState({
-                          query: query
-                      });
+            query: query
+        });
     }
 
     doSearch() {
-        if (this.state.type || this.state.query) {
+        if (this.state.query) {
+            let query = this.state.query.toLowerCase();
+            const options = {keywords: ['uid', 'type', 'tag', 'query', 'moduleid', 'status', 'name'], offsets: false};
+            //Basic Regex Matching to fix query inconsistencies
+            //Removing all extra spaces i.e. two or more
+            query = query.replace(/\s+/g,' ').trim();
+            //Removing all spaces to the left of search keys
+            query = query.replace(new RegExp("\\s+:","g"),":");
+            //Removing all spaces to the right of search keys
+            query = query.replace(new RegExp(":\\s+","g"),":");
             let url = "/search?";
-            if (this.state.type) {
-                url += "type=" + this.state.type;
+            // Check if query is an advanced query. We want to use the Parser only if the query is advanced
+            if (query.includes(":") || query.includes("-")) {
```

Comment:
Peeking at https://www.npmjs.com/package/search-query-parser it says

> If no keywords or ranges are specified, or if none are present in the given search query, then searchQuery.parse will return a string if tokenize is false

so you should rely on this, and avoid doing this test yourself. (E.g. what if I query for: notAValidKey:foo ?)

Reply:
@djazayeri , although from what's mentioned in the docs, it seems that the parser just returns a string, it actually returns something like query: "user query" . This makes it impossible to differentiate a normal query from an advanced one. That is why I had implemented this test to check the pre-parsed query